src/ucp/wireup/select.c

```diff
@@ -889,9 +889,14 @@ static UCS_F_NOINLINE ucs_status_t ucp_wireup_add_memaccess_lanes(
         ucp_tl_bitmap_t tl_bitmap, ucp_lane_type_t lane_type,
         ucp_wireup_select_context_t *select_ctx)
 {
+    ucp_context_h context                = select_params->ep->worker->context;
     ucp_wireup_criteria_t mem_criteria   = *criteria;
     ucp_wireup_select_info_t select_info = {0};
-    int show_error                       = !select_params->allow_am;
+    int allow_am                         =
+            ((lane_type == UCP_LANE_TYPE_RMA) &&
+             (context->config.features & UCP_FEATURE_EXPORTED_MEMH)) ?
+            0 : select_params->allow_am;
```

Comment:
1. can use && expression instead of ?:
2. since it's long now, better not initialize it during declaration
3. what about atomics?

Reply:
> can use && expression instead of ?:

not sure that I understand

> since it's long now, better not initialize it during declaration

done

> what about atomics?

EXPORTED_MEMH is not supported, because AMO doesn't use memh from request's params

Reply:
```c
allow_am = select_params->allow_am &&
       !((lane_type == UCP_LANE_TYPE_RMA) &&
        (context->config.features & UCP_FEATURE_EXPORTED_MEMH));
```